proposals/Exceptions.md

```diff
@@ -170,21 +174,101 @@ also be sure to also pop off the caught exception values.
 
 ### Rethrows
 
-The _rethrow_ instruction has no arguments, and can only appear in catch
-blocks. It always re-throws the exception caught by the catch block. This allows
-the catch block to clean up state before the exception is passed back to the
-next enclosing try block.
+The _rethrow_ instruction can only appear in the body of a catch block.  The
+_rethrow_ instruction always re-throws the exception caught by the referenced
+catch block. This allows the catch block to clean up state before the exception
+is passed back to the next enclosing try block.
+
+The _rethrow_ instruction gets a _catch index_ argument. It is to disambiguate
+which caught exception is to be rethrown, when inside nested catch blocks (of
+corresponding nested try blocks).
+
+For example consider the following example:
+
+```
+try
+  ...
+catch 1
+  ...
+  try
+    ...
+  catch 2
+    ...
+    try
+      ...
+    catch 3
+      ...
+      rethrow N
+    end
+  end
+end
+```
+
+In this example, `N` is used to disambiguate which caught exception is being
+rethrown. It could rethrow any of the three caught expceptions. The catch blocks
+are numbered monotonically (inside-out) starting at zero. Zero implies the
+immediately enclosing catch block.
+
+So, in the above example, `rethrow 0` corresponds to the exception caught by
+`catch 3`, `rethrow 1` corresponds to the exception caught by `catch 2`, and
+`rethrow 2` corresponds to the exception caught by `catch 1`.
 
 ### Debugging
 
 Earlier discussion implied that when an exception is thrown, the runtime will
-pop the value stack across function calls until a corresponding, enclosing try
+pop the operand stack across function calls until a corresponding, enclosing try
 block is found. The implementation may actually not do this. Rather, it may
 first search up the call stack to see if there is an enclosing try. If none are
 found, it could terminate the thread at that point of the throw. This would
 allow better debugging capability, since the corresponding call stack is still
 there to query.
 
+## Changes to the text format.
+
+This section describes change in the
+[instruction syntax document](https://github.com/WebAssembly/spec/blob/master/document/syntax/instructions.rst).
```

Comment:
Note that this points to the abstract syntax, not the text format. The latter will be under https://github.com/WebAssembly/spec/blob/master/document/text/instructions.rst, once spec PR #471 has landed.